internal/quaiapi/api.go

```diff
@@ -1233,15 +1230,6 @@ func AccessList(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrH
 		b.Logger().WithField("accessList", accessList).Debug("Creating access list")
 		// Set the accesslist to the last al
 		args.AccessList = &accessList
-		// If no gas amount was specified, each unique access list needs it's own
-		// gas calculation. This is quite expensive, but we need to be accurate
-		// and it's convered by the sender only anyway.
-		if nogas {
-			args.Gas = nil
-			if err := args.setDefaults(ctx, b, db); err != nil {
```

Comment:
you still want to call this setDefaults right? because thats the function you changed

Reply:
We only have to call it once, don’t have to do it in the for loop